### PR TITLE
Fix 0103 warning for non-injected pods on the host network

### PR DESCRIPTION
**Pull request by cnvergence**
**Please provide a description of this PR:**
This fixes a warning that shows only on injection-enabled namespaces with deployed pods on the host network.
Automatic sidecar injection is disabled for such pods by default.

### CODE REVIEW

**Review by GregHanson — CHANGES_REQUESTED** [review]
@cnvergence could you add a [release note](https://github.com/istio/istio/tree/master/releasenotes) for this fix?

**Review by howardjohn — APPROVED** [review]

File: releasenotes/notes/40220.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+releaseNotes:
+  - |
+    **Fixed** IST0103 warning from `istioctl analyze` for not injected pods on the host network.
```
Comment (ericvn) [inline releasenotes/notes/40220.yaml:6]: c/not injected/non-injected/

**Review by ericvn — APPROVED** [review]
RN looks good.

**Comment by cnvergence** [comment]
@GregHanson PTAL

**Pull request merged**